### PR TITLE
Upgraded Run Script Too

### PR DESCRIPTION
**Pull request by KailHet**
Changes:
1. Option to Hide/Show code field
2. New function to save variables from code (also have docs for it)
3. Updated style of the mod

Mod by Jek'a

### CODE REVIEW

**Review by DetroitWhiskey136 — CHANGES_REQUESTED** [review]
1) why did you remove the meta field?
2) you have a bunch of linting/prettier errors.
3) under fields try to use human readable variable names.

**Review by DetroitWhiskey136 — COMMENTED** [review]
For the linting errors regarding the unexpected functions, you can use an inline comment to ignore them. You can also place an underscore at the beginning of function arguments to signify that the argument isn't going to be used,
EXAMPLE:
`// eslint-disable-next-line func-names` ~ ignoring unnamed functions.
`glob.onDTLChanged = function (_event) {` ~ ignoring the fact event is never used.

File: actions/run_script_too_MOD.js [hunk 1]
```diff
@@ -19,107 +18,215 @@ module.exports = {
...
+    <textarea id="code" value="console.log("Text!")" rows="14" name="is-eval" style="width: 99%; white-space: nowrap; resize: none;"></textarea>
+  </div>
+  </div></div><br><br>`;
   },
 
-  init() {},
+  init() {
+    const { glob, document } = this;
+
+    glob.onSVarChanged = function (event) {
```
Comment (DetroitWhiskey136) [inline actions/run_script_too_MOD.js:166]:
```suggestion
    // eslint-disable-next-line func-names
    glob.onSVarChanged = function (event) {
```

File: actions/run_script_too_MOD.js [hunk 2]
```diff
@@ -19,107 +18,215 @@ module.exports = {
...
+        document.getElementById('svars').style.display = 'none';
+      } else {
+        document.getElementById('svars').style.display = null;
+      }
+    };
+
+    glob.onSVarChanged(document.getElementById('savevariables'));
 
-  async action(cache) {
+    glob.onShowCodeTypeChanged = function (event) {
```
Comment (DetroitWhiskey136) [inline actions/run_script_too_MOD.js:176]:
```suggestion
    // eslint-disable-next-line func-names
    glob.onShowCodeTypeChanged = function (event) {
```

File: actions/run_script_too_MOD.js [hunk 3]
```diff
@@ -19,107 +18,215 @@ module.exports = {
...
+      if (event.value === '0') {
+        document.getElementById('customcode').style.display = 'none';
+      } else {
+        document.getElementById('customcode').style.display = null;
+      }
+    };
+
+    glob.onShowCodeTypeChanged(document.getElementById('showcode'));
+
+    glob.onDTLChanged = function (event) {
```
Comment (DetroitWhiskey136) [inline actions/run_script_too_MOD.js:186]:
```suggestion
     // eslint-disable-next-line func-names
    glob.onDTLChanged = function (_event) {
```

**Review by DetroitWhiskey136 — CHANGES_REQUESTED** [review]

**Comment by OneAndonlyFinbar** [comment]
Closing because OP seems to have gone inactive. Feel free to reopen later.

**Pull request closed without merging**